donkeycar/pipeline/types.py

```diff
@@ -32,36 +32,41 @@ def __init__(self, config: Any, base_path: str, underlying: TubRecordDict) -> No
         self.underlying = underlying
         self._image: Optional[Any] = None
 
-    def image(self, cached=True, normalize=False) -> np.ndarray:
-        if not self._image:
+    def image(self, cached=True) -> np.ndarray:
+        if self._image is None:
             image_path = self.underlying['cam/image_array']
             full_path = os.path.join(self.base_path, 'images', image_path)
             _image = load_image_arr(full_path, cfg=self.config)
-            if normalize:
-                _image = normalize_image(_image)
             if cached:
                 self._image = _image
-            return _image
         else:
-            return self._image
+            _image = self._image
+        return _image
 
     def __repr__(self) -> str:
         return repr(self.underlying)
 
 
 class TubDataset(object):
-    def __init__(self, paths: List[str], config: Any) -> None:
-        self.paths = paths
+    '''
+    Loads the dataset, and creates a train/test split.
+    '''
+
+    def __init__(self, config: Any, tub_paths: List[str], shuffle: bool = True):
         self.config = config
-        self.tubs = [Tub(path) for path in self.paths]
+        self.tub_paths = tub_paths
+        self.shuffle = shuffle
+        self.tubs: List[Tub] = [Tub(tub_path, read_only=True)
+                                for tub_path in self.tub_paths]
         self.records: List[TubRecord] = list()
 
-    def load_records(self) -> List[TubRecord]:
-        self.records.clear()
+    def train_test_split(self) -> Tuple[List[TubRecord], List[TubRecord]]:
+        print(f'Loading tubs from paths {self.tub_paths}')
```

Comment:
You should call `self.records.clear()` as the first step just in case someone calls this method repeatedly.

Reply:
Yes, good point. Done.